src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp

```diff
@@ -340,16 +327,15 @@ StatefulSDPAFusion::StatefulSDPAFusion() {
 
 bool SDPASubgraphFusion::run_on_model(const std::shared_ptr<ov::Model>& f) {
     RUN_ON_FUNCTION_SCOPE(SDPASubgraphFusion);
-    ov::pass::Manager manager("SDPASubgraphFusion");
+    ov::pass::SymbolicOptimizations symbolic_optimizations(false, get_pass_config());
+    auto& ctx_manager = *symbolic_optimizations.get_manager();
```

Comment:
Why we can't use the shared pointer here? it looks a bit dangerous

Reply:
lets move it into the transformation like we did heir - https://github.com/openvinotoolkit/openvino/pull/31440

Reply:
I use #if defined(OPENVINO_ARCH_X86_64) instead of CPU_REGISTER_PASS_X64

Reply:
The macro could be used here to register pass conditionally.

Using "auto& ctx_manager = *symbolic_optimizations.get_manager();" should be fine as the reference is local and there is no need to use shared pointer (it would faster as there is no need to be copied) and we are sure object will exist here (unless I've missed something).

What about use:
```cpp
    CPU_REGISTER_PASS_X64((*ctx_manger), ov::intel_cpu::SDPAFuseTransposeReshape));
```
instead of:
```cpp
#if defined(OPENVINO_ARCH_X86_64)
    ctx_manager->register_pass<ov::intel_cpu::SDPAFuseTransposeReshape>();
#endif
```